### DISCUSSION TITLE
Save to include videos/gifs

### DISCUSSION THREAD

**Discussion by megamorphg** [gildas-lormeau/SingleFile#1432] [Q&A]
**Is your feature request related to a problem? Please describe.**
I would like to save a [page](https://www.bachatasteps.com/?options=level) that contains many dance clips in gif format.
When saving the page it does not include all the images or any videos. At best, there are just some static images.
I've enabled all options under Images section including setting the maximum idle time to 10s.

**Describe the solution you'd like**
Way to pre-load all the videos that are visible and save as a single file.

**Describe alternatives you've considered (optional)**
I'm open to any other alternatives! Would just like to be able to access the site offline.

**Comment by gildas-lormeau** [comment]
Actually, videos can be saved if you uncheck the option `Network > blocked resources > videos` but they cannot be played because of an HTML element placed on top of them and preventing you to use the context menu. 
To circumvent this, you can press F12 to open the browser debugger just before saving the page, click on the "Console" tab, and type the code below in the console.
```js
document.querySelectorAll(".video-overlay").forEach(overlayElement => overlayElement.remove())
```

FYI, here's the saved page with this changes applied before saving it: [Level_Difficulty _ BachateSteps.com _ Bachata Database with 900_ moves_combos! (4_4_2024 2_48_53 PM).html.zip](https://github.com/gildas-lormeau/SingleFile/files/14871524/Level_Difficulty._.BachateSteps.com._.Bachata.Database.with.900_.moves_combos.4_4_2024.2_48_53.PM.html.zip)

**Comment by megamorphg** [comment]
Awesome, that worked! Would you happen to know how to make all the videos play? It seems even when I allow javascript to download all the videos don't play simultaneously offline as when loading the page online.

**Comment by gildas-lormeau** [comment]
Run the code instead to also autoplay all the videos. Note that you could convert it to a [bookmarklet](https://en.wikipedia.org/wiki/Bookmarklet) for more convenience.

```js
document.querySelectorAll(".video-overlay").forEach(overlayElement => overlayElement.remove()); document.querySelectorAll("video").forEach(videoElement => videoElement.autoplay = true);
```

**Reply by bennjmin** [reply]
It seems a bit ridiculous to me that this solution would not be built into the SingleFile extension itself. I'm just speaking from a practical point of view: SingleFile is awesome and easy to use - but then coming up with extra pieces of code to execute somewhere... No, not really.
